### ISSUE TITLE
Syncing previous reviews from mobile makes Straight Reward move them to Relearning

### ISSUE THREAD

**Issue by kelpso** [hgiesel/anki_straight_reward#19]
Scenario:
1. Do 75% of cards on AnkiDroid.
2. Sync to AnkiWeb.
3. Do remaining 25% of cards on AnkiDroid.
4. Go to sleep without syncing
5. Wake up and sync on Anki desktop
6. Realise you forgot to sync the last 25% on AnkiDroid
7. Sync on AnkiDroid
8. Sync on Anki desktop

![image](https://user-images.githubusercontent.com/13115959/124370130-31596480-dcb7-11eb-8976-820821179e14.png)

As in the image above, all 25% have kept their review history from last night in the stats, but have all been marked as relearning and due today so it's like last night's review never happened. Reviewing again will make Anki calculate due date accordingly, so for a card that was New last night, it will be set to 1 day after today's review (basically reviewing the same card three days in a row) as per the image below.

![image](https://user-images.githubusercontent.com/13115959/124370133-374f4580-dcb7-11eb-83d4-5de7825bd397.png)

Only fix is to set the clock back to last night and blast through the 25% of cards again so that Anki recalculates the due date. But then review history looks as below, with two reviews on the same day. This isn't so big of a problem but would be better if it didn't happen at all.

![image](https://user-images.githubusercontent.com/13115959/124370138-40401700-dcb7-11eb-8988-4c861f1ecde0.png)

25% and 75% are just made-up numbers, you can do this with any number of cards and it will still occur. Also this is with default scheduling settings on Anki and next day starts at hours past midnight set to 5.

**Comment by hgiesel** [comment]
I don't think this is an issue with the add-on? The revlog is append-only.

**Comment by kelpso** [comment]
I don't use any other add-ons that mess with the scheduling/ease, so either it's a problem with Anki or it's a problem with the add-on. If it's a problem with Anki I'm not sure I could convince the team to fix it if the bug only occurs while using this add-on.

Did you attempt to reproduce the issue?

**Comment by hgiesel** [comment]
Honestly I don't understand the issue. However what you're showing me is the revlog, and the sync-hook doesn't meddle with the revlog at all.

**Comment by kelpso** [comment]
The issue is syncing in on the Straight Rewards device while there is still unsynced progress on another device, then syncing out that other device to bring the remaining progress into the Straight Rewards device makes it reset the card states to learning while still keeping the review history intact. I can report it as an Anki bug but the issue doesn't occur without Straight Rewards installed, so it might not be directly caused by the add-on but it's possibly an issue with how Anki manages its database integrity and its relation to the ease which is uncovered by the add-on.

**Comment by kuroahna** [comment]
I've experienced this issue as well only after installing Straight Rewards. I didn't know how to reproduce it but the addon seems to be the cause.

**Comment by kelpso** [comment]
Not sure if this helps but I wrote up an example with times which might help visualise what seems to be happening:

1. Last sync in Anki desktop was 12:00 pm
2. I reviewed a card on AnkiDroid at 11:50 am but didn't sync it
3. I sync AnkiDroid at 12:05 pm
4. Straight Rewards sees it last synced at 12:00 pm so messes up my 11:50 am card because it has no record of that review in its history

I'll post this on the Anki forum a bit later today as well just to make sure.
